### ISSUE TITLE
Link to favicon not included in HTML when logged in

### ISSUE THREAD

**Issue by Dravere** [NodeBB/NodeBB#6518]
- **NodeBB version:** 1.9.2
- **NodeBB git hash:** 1233730
- **Database type:** mongo
- **Database version:** 3.6.4
- **Exact steps to cause this issue:**
  * Go to your NodeBB site (private window, so you're not logged in)
  * Press F12 and look at the source code. There should be a `<link rel="icon" type="image/x-icon" href="/forum/favicon.ico?v=2amjbe4vk3t" />` or similar in the `head` part of the HTML.
  * Log in
  * Press F12 and look at the source code. There is no more favicon in the `head` part.
- **What you expected:** I expected that the favicon link would be available independent of being logged in or not.
- **What happened instead:** The favicon link disappeared. And the browser then takes another available image (in my case I think it is the OpenGraph image) scaling it down in a horrible way.

I tried to find out why this is happening. But I have no clue. I found the `meta.tags.parse` and the hook `filter:meta.getMetaTags`. But after the hook is fired I can still see the icon tag in there. So it probably is removed somewhere later? But when and where and why? Especially the strange difference between being logged in and not?

**Comment by barisusakli** [comment]
Looks like https://github.com/NodeBB/NodeBB/blob/master/public/src/modules/notifications.js#L175 that line is causing the link tag in head to disappear, if I comment that out I can see the tag.

**Comment by Dravere** [comment]
Oh, now I get it. The link is still there but at the very end of the head now and a data url. And it isn't the problem that it gets removed. The problem is that I'm using a multi layer ico image for different resolution. Tinycon is unable to work with that. It doesn't take the most appropriate size and just destroys the icon during resizing.

**Comment by julianlam** [comment]
Yes, unfortunately I think that's an issue with Tinycon. I did upgrade it awhile back so that it saves the icon in a higher resolution, although I don't think it handles multi-layer icons...

**Comment by Dravere** [comment]
I wonder if they can fix it. There isn't really an easy fix to this. The only possibility I see is to use an Ico file parser, extract the right resolution and work with that. It seems to be impossible to do this with an image element. That is quite a bit of work to just display a tiny icon.

Would it be an option to provide a switch in NodeBB to set the fallback method of Tinycon to `force`? Thus only adding the notification to the title like for example Twitter does?

**Comment by julianlam** [comment]
Would probably want to switch to a different "notification counter in favicon" lib if there is a good one

**Comment by mySebbe** [comment]
The original mechanism is unchanged: current Core still uses Tinycon and calls `setBubble()` for notification counts, while ACP exposes only the bubble colors. There is no opt-out or title-only mode for multi-layer ICO files. Would maintainers prefer a small compatibility setting that disables favicon mutation (leaving the existing in-page/app badge), or should this wait for replacement of Tinycon? Those are different scopes, and the former is much easier to regression-test.
